src/main/java/org/jenkinsci/backend/confluence/pageremover/webapp/PageRemoverResource.java

```diff
@@ -0,0 +1,189 @@
+import hudson.plugins.jira.soap.RemotePage;
+import org.apache.commons.mail.Email;
+import org.apache.commons.mail.EmailException;
+import org.apache.commons.mail.SimpleEmail;
+import org.jenkinsci.backend.confluence.pageremover.Connection;
+import org.jenkinsci.backend.confluence.pageremover.LanguageDetection;
+import org.jenkinsci.backend.confluence.pageremover.PageNotification;
+import org.jenkinsci.backend.confluence.pageremover.Space;
+import org.jenkinsci.backend.confluence.pageremover.Spambot;
+import org.jenkinsci.backend.confluence.pageremover.WordList;
+import org.jenkinsci.backend.ldap.AccountServer;
+import org.jooq.lambda.Unchecked;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.apache.commons.lang.exception.ExceptionUtils.getFullStackTrace;
+import static org.jenkinsci.backend.confluence.pageremover.Spambot.BLACKLIST;
+
+@Path("/page-remover")
+@Produces(MediaType.TEXT_PLAIN)
+public class PageRemoverResource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PageRemoverResource.class);
+
+    private final String mailRecipient;
+    private final String smtpServer;
+    private final AccountServer accountServer;
+    private final List<String> languagesToBlock;
+    private final float blockingProbability;
+
+
+    public PageRemoverResource(String mailRecipient, String smtpServer,
+                               AccountServer accountServer, List<String> languagesToBlock,
+                               float blockingProbability) {
+        this.mailRecipient = mailRecipient;
+        this.smtpServer = smtpServer;
+        this.accountServer = accountServer;
+        this.languagesToBlock = languagesToBlock;
+        this.blockingProbability = blockingProbability;
+    }
+
+    @POST
+    public Response submit(@QueryParam("subject") Optional<String> subject,
+                           @QueryParam("body-plain") Optional<String> body) throws Exception {
+
+        PageNotification pageNotification = PageNotification.parse(subject.or(""), body.or(""));
+
+        if (subject.or("").startsWith("Re: ") && body.isPresent()) {
+            removePage(subject.get(), body.get());
+            return Response.ok().build();
+        }
+
+        if (pageNotification != null) {
+            processNotification(pageNotification);
+            return Response.ok().build();
+        }
+
+        LOGGER.warn("Not a confluence notification email: " + subject.or("[empty subject]"));
+        return Response.ok().build();
+
+    }
+
+    /**
+     * Processes notification email from Confluence, and take appropriate actions.
+     */
+    private void processNotification(PageNotification n) throws Exception {
```

Comment:
Can't we just reuse the code we already have for doing this logic instead of copying it here?